src/helpers/with-middleware-auth-required.ts

```diff
@@ -0,0 +1,88 @@
+import { NextMiddleware, NextRequest, NextResponse } from 'next/server';
+import { SessionCache } from '../session';
+
+/**
+ * Protect your pages with Next.js Middleware. eg
```

Comment:
```suggestion
 * Protect your pages with Next.js Middleware. For example:
```
The docs guidelines suggest not to use latin abbreviations.